L1Trigger/L1TMuonOverlapPhase2/src/PtAssignmentNNRegression.cc

```diff
@@ -0,0 +1,308 @@
+    }
+    return true;
+  }
+
+  return false;
+}
+
+PtAssignmentNNRegression::~PtAssignmentNNRegression() {
+  // TODO Auto-generated destructor stub
+}
+
+std::vector<float> PtAssignmentNNRegression::getPts(AlgoMuons::value_type& algoMuon,
+                                                    std::vector<std::unique_ptr<IOMTFEmulationObserver>>& observers) {
+  LogTrace("l1tOmtfEventPrint") << " " << __FUNCTION__ << ":" << __LINE__ << std::endl;
+  auto& gpResult = algoMuon->getGpResultConstr();
+  //int pdfMiddle = 1<<(omtfConfig->nPdfAddrBits()-1);
+
+  LogTrace("l1tOmtfEventPrint") << " " << __FUNCTION__ << ":" << __LINE__ << std::endl;
+  /*
+  edm::LogVerbatim("l1tOmtfEventPrint")<<"DataROOTDumper2:;observeEventEnd muonPt "<<event.muonPt<<" muonCharge "<<event.muonCharge
+      <<" omtfPt "<<event.omtfPt<<" RefLayer "<<event.omtfRefLayer<<" omtfPtCont "<<event.omtfPtCont
+      <<std::endl;
+*/
+
+  unsigned int inputCnt = 18;  //TDOO!!!!!
+  unsigned int outputCnt = 2;
+  const float noHitVal = 1023.;
+
+  //edm::LogImportant("OMTFReconstruction") <<"\n----------------------"<<endl;
+  //edm::LogImportant("OMTFReconstruction") <<(*algoMuon)<<std::endl;
+
+  std::vector<float> inputs(inputCnt, noHitVal);
+
+  for (unsigned int iLogicLayer = 0; iLogicLayer < gpResult.getStubResults().size(); ++iLogicLayer) {
+    auto& stubResult = gpResult.getStubResults()[iLogicLayer];
+    if (stubResult.getMuonStub()) {  //&& stubResult.getValid() //TODO!!!!!!!!!!!!!!!!1
+      int hitPhi = stubResult.getMuonStub()->phiHw;
+      unsigned int refLayerLogicNum = omtfConfig->getRefToLogicNumber()[algoMuon->getRefLayer()];
+      int phiRefHit = gpResult.getStubResults()[refLayerLogicNum].getMuonStub()->phiHw;
+
+      if (omtfConfig->isBendingLayer(iLogicLayer)) {
+        hitPhi = stubResult.getMuonStub()->phiBHw;
+        phiRefHit = 0;  //phi ref hit for the banding layer set to 0, since it should not be included in the phiDist
+      }
+
+      OmtfHit hit(0);
+
+      hit.layer = iLogicLayer;
+      hit.quality = stubResult.getMuonStub()->qualityHw;
+      hit.eta = stubResult.getMuonStub()->etaHw;  //in which scale?
+      hit.valid = stubResult.getValid();
+
+      //phiDist = hitPhi - phiRefHit;
+      hit.phiDist = hitPhi - phiRefHit;
+
+      /*
+      LogTrace("l1tOmtfEventPrint") <<" muonPt "<<event.muonPt<<" omtfPt "<<event.omtfPt<<" RefLayer "<<event.omtfRefLayer
+          <<" layer "<<int(hit.layer)<<" PdfBin "<<stubResult.getPdfBin()<<" hit.phiDist "<<hit.phiDist<<" valid "<<stubResult.getValid()<<" " //<<" phiDist "<<phiDist
+          <<" getDistPhiBitShift "<<omtfCand->getGoldenPatern()->getDistPhiBitShift(iLogicLayer, omtfCand->getRefLayer())
+          <<" meanDistPhiValue   "<<omtfCand->getGoldenPatern()->meanDistPhiValue(iLogicLayer, omtfCand->getRefLayer())//<<(phiDist != hit.phiDist? "!!!!!!!<<<<<" : "")
+          <<endl;*/
+
+      if (hit.phiDist > 504 || hit.phiDist < -512) {
```

Comment:
Could these numbers be a bit better explained with some constants?

Reply:
@kbunkow, can you take a look, please?